File: Answer/1161.最大层内元素和.go
/*
 * @lc app=leetcode.cn id=1161 lang=golang
 *
 * [1161] 最大层内元素和
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxLevelSum(root *TreeNode) int {
	max := 0
	sum := []int{}

	var dfs func(*TreeNode, int)
	dfs = func(root *TreeNode, level int) {
		if len(sum) > level {
			sum[level] += root.Val
		} else {
			sum = append(sum, root.Val)
		}
		if root.Left != nil {
			dfs(root.Left, level+1)
		}
		if root.Right != nil {
			dfs(root.Right, level+1)
		}
	}

	dfs(root, 0)
	for i := 0; i < len(sum); i++ {
		if sum[i] > sum[max] {
			max = i
		}
	}
	return max + 1
}

// @lc code=end

File: Answer/1161.最大层内元素和.js
/*
 * @lc app=leetcode.cn id=1161 lang=javascript
 *
 * [1161] 最大层内元素和
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxLevelSum = function (root) {
    let max = 0;
    let sum = Array();
    Dfs(root, 0, sum);
    for (let i = 0; i < sum.length; i++) {
        if (sum[i] > sum[max]) {
            max = i;
        }
    }
    return max + 1;
};

/**
 * @param {TreeNode} root
 * @param {number} level
 * @param {Array} sum
 */
function Dfs(root, level, sum) {
    if (root != null) {
        if (sum.length > level) {
            sum[level] += root.val
        } else {
            sum.push(root.val);
        }
        Dfs(root.left, level + 1, sum);
        Dfs(root.right, level + 1, sum);
    }
}
// @lc code=end

